### DISCUSSION TITLE
Acquiring Stress Tensors per Grain

### DISCUSSION THREAD

**Discussion by dannypanduro** [neperfepx/FEPX#14] [Q&A] [answered]
Hello All,

I hope you're all doing well.
I was wondering if it was possible to generate stress tensors for a single grain instead of elements? 

Thank you,

Richard Panduro

**Comment by mkasemer** [comment, marked as answer]
Hi @RichardPA,

FEPX does not output grain-averaged values. This is a post-processing operation, and can possibly be done via Neper (please post in the discussion forum there for guidance).

You can also calculate externally: The mesh contains information about which elements belong to which grains, and you can print element volumes at every step. Volume averaging on a per-grain basis would be a relatively easy calculation given this information.

**Reply by dannypanduro** [reply]
Hi @mkasemer ,

I will explore the Neper manual, and look into these concepts. If I can't figure it out, I'll post in the Neper discussion forum. 

Thank you!

Richard Panduro

**Reply by rquey** [reply]
Short reply (even if this is Neper stuff): `neper -S -elsetres` is exactly what you are looking for (`neper -S` reformats the raw FEPX results but also includes post-processing operations such as this one). As @mkasemer said, please post to the Neper discussion forum for more information.

**Reply by dannypanduro** [reply]
Thank you @rquey, I haven't used the -S module much and it skipped my brain. I appreciate the solution! Also, I apologize, I didn't realize this was a Neper question/solution and I'll post questions about postprocessing in the Neper discussion in the future.

Thank you!
